### PR TITLE
feat: global ⌘K command palette + punctuation-insensitive search everywhere / 全局 ⌘K 命令面板与全站忽略标点搜索

### LINKED ISSUE #406: Search optimizations (Not SEO) / 搜索优化（非 SEO）
Image

When searching for GPU configs, it shld be parenthesis-less, where "B300 vllm" search would have "B300 (vllm)" show up

## 中文说明
搜索 GPU 配置时应忽略括号，例如输入"B300 vllm"应能匹配到"B300 (vllm)"等结果。

### PR DESCRIPTION
**Pull request by functionstackx**
## Summary

Two usability upgrades aimed at making the app dramatically faster to navigate and search:

### 1. Global command palette (⌘K / Ctrl+K)

A keyboard-first launcher in the header that jumps to **any destination in the app** from anywhere:

- All top-level pages (Home, AgentX, Overview, Comparisons, Articles, About, AI Chips, Rankings, Glossary, Supporter Quotes, API Reference)
- All primary dashboard tabs (Inference Performance, Accuracy Evals, Historical Trends, TCO Calculator, Fleet Lifecycle, Chip Specs, Submissions)
- Every active model dashboard (`/inference/<model>`)
- Every chip page (`/chips/<slug>`)
- Quick actions: switch theme, switch language, star on GitHub

Details:

- Fully bilingual — on `/zh` pages the palette renders Chinese labels and navigates to the `/zh` sibling routes
- Full keyboard support (↑/↓/Home/End/Enter/Esc), ARIA combobox/listbox semantics, `aria-activedescendant`, scroll-into-view
- Uses the same commit-retry navigation as header links (extracted as `pushInApp` from `navigateInApp`) so the first dashboard transition commits reliably
- PostHog events: `command_palette_opened` / `command_palette_selected`
- Trigger button in the header with platform-aware ⌘K / Ctrl K hint (rendered post-mount for hydration safety)

### 2. Punctuation-insensitive, order-insensitive search everywhere — closes #406

New shared `matchesSearch` utility (`packages/app/src/lib/search-match.ts`): lowercases, folds separator punctuation to spaces (CJK preserved), and requires every query token to match. Applied to:

- `SearchableSelect` (model/queue pickers) — "B300 vllm" now matches "B300 (vLLM)"
- `MultiSelect` (both filter paths)
- `DataTable` global filter
- The new command palette

### Tests

- `search-match.test.ts` — normalization, tokenization, #406 regression cases, CJK
- `command-palette-items.test.ts` — registry pinned against `chip-pages.ts`, dashboard routes, and model slugs so the palette can't silently drift
- `command-palette.test.tsx` — open via trigger/Ctrl+K, filtering, Enter navigation, arrow keys, empty state, `/zh` behavior
- `searchable-select.test.ts` — new parenthesis/word-order case

`bun run typecheck` ✓ · `bun run lint` ✓ · `bun run fmt` ✓ · `bun run test:unit` ✓ (254 files, 4486 tests)

Verified in a live dev server (desktop 1440px, mobile 375px, `/` and `/zh`): palette opens via shortcut and button, filters, navigates to pages/tabs/models/chips, and the header hint renders on one line.

## 中文说明

两项可用性升级，让站内导航与搜索显著提速：

### 1. 全局命令面板（⌘K / Ctrl+K）

页眉新增键盘优先的启动器，可从任意位置跳转到应用内所有目标：全部顶层页面、主要仪表板标签页（推理性能、精度评测、历史趋势、TCO 计算器、机队生命周期、芯片规格、提交记录）、每个模型仪表板（`/inference/<model>`）与每个芯片页面（`/chips/<slug>`），并提供快捷操作（切换主题、切换语言、GitHub 加星）。

- 完整双语支持——在 `/zh` 页面上面板显示中文标签，并跳转到对应的 `/zh` 路由
- 完整键盘支持（↑/↓/Home/End/Enter/Esc）与 ARIA combobox/listbox 语义
- 复用页眉链接的导航重试逻辑（从 `navigateInApp` 提取为 `pushInApp`），确保首次仪表板跳转可靠提交
- PostHog 事件：`command_palette_opened` / `command_palette_selected`

### 2. 全站搜索忽略标点与词序——修复 #406

新增共享工具 `matchesSearch`：转小写、将分隔标点折叠为空格（保留中文字符）、要求每个查询词均命中。已应用于下拉搜索、多选、数据表格全局筛选与新命令面板。例如 "B300 vllm" 现可匹配 "B300 (vLLM)"。

### 测试

新增/扩展四组单元测试；`typecheck`、`lint`、`fmt`、`test:unit`（254 个文件，4486 个用例）全部通过，并在本地开发服务器上完成桌面端、移动端与中文站的可视化验证。

---

> [!NOTE]
> **Medium Risk**
> Touches app-wide navigation and in-app routing retry logic, plus search behavior in shared UI controls; risk is mitigated by extensive unit tests but edge cases around dashboard URL state remain possible.
> 
> **Overview**
> Adds a **global command palette** (header trigger + **⌘K / Ctrl+K**) that searches and jumps to site pages, dashboard tabs, model routes, and chip pages, with quick actions for theme, locale, and GitHub. Navigation reuses the header’s **commit-retry** behavior via new **`pushInApp`**, preserves dashboard query params (e.g. `unofficialruns`), skips re-navigating to the current route, and is bilingual on **`/zh`**. The palette trigger is hidden below **360px** width to keep narrow headers usable.
> 
> Introduces shared **`matchesSearch`** (token-based, punctuation-normalized, order-independent) and wires it into **SearchableSelect**, **MultiSelect**, **DataTable** filtering, and the palette—fixing **#406** (e.g. `B300 vllm` matches `B300 (vLLM)`).
> 
> **`TAB_LABELS_EN`** moves to **`tab-meta.ts`** for reuse by tab nav and palette; palette destinations live in test-backed **`command-palette-items`**. Unit tests cover search matching, palette behavior, and registry drift vs chip pages and model slugs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 275bec50f59077f3b05932e87535a4dac9463212. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/app/src/components/ui/command-palette.tsx [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+      setActiveIndex(0);
+      scrollRowIntoView(0);
+    } else if (event.key === 'End' && flatEntries.length > 0) {
+      event.preventDefault();
+      setActiveIndex(flatEntries.length - 1);
+      scrollRowIntoView(flatEntries.length - 1);
+    } else if (event.key === 'Enter') {
+      event.preventDefault();
+      flatEntries[clampedIndex]?.select();
+    }
```
[the remark this answers is not in the document]
Comment (functionstackx) [reply packages/app/src/components/ui/command-palette.tsx:297]: Fixed in 6f06cbd — Enter is now ignored while an IME composition is being confirmed (`event.nativeEvent.isComposing || event.keyCode === 229`), with a regression test.

File: packages/app/src/components/ui/command-palette.tsx [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+        },
+      },
+      {
+        id: 'action:locale',
+        label: t.switchLocale,
+        keywords: t.switchLocaleKeywords,
+        icon: LanguagesIcon,
+        run: () => {
+          router.push(switchLocalePath(pathname));
+        },
```
[the remark this answers is not in the document]
Comment (functionstackx) [reply packages/app/src/components/ui/command-palette.tsx:210]: Fixed in 6f06cbd — the locale action now appends the live query string (`useClientSearch`) and navigates via `pushInApp`, matching the header language toggle.

File: packages/app/src/components/ui/command-palette.tsx [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+        event.preventDefault();
+        setOpen((prev) => {
+          if (!prev) track('command_palette_opened', { source: 'shortcut' });
+          return !prev;
+        });
+      }
+    };
+    document.addEventListener('keydown', onKeyDown);
+    return () => document.removeEventListener('keydown', onKeyDown);
+  }, []);
```
[the remark this answers is not in the document]
Comment (functionstackx) [reply packages/app/src/components/ui/command-palette.tsx]: Fixed in 6f06cbd — the shortcut toggle now routes through `handleOpenChange`, so closing via ⌘K/Ctrl+K clears the query and active row; covered by a new test.

File: packages/app/src/components/ui/command-palette.tsx [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+  }, [handleOpenChange]);
+
+  const navItems = React.useMemo(() => buildPaletteNavItems(locale), [locale]);
+
+  const selectNav = React.useCallback(
+    (item: PaletteNavItem) => {
+      const target = locale === 'zh' && hasZhSibling(item.href) ? zhPath(item.href) : item.href;
+      track('command_palette_selected', { id: item.id, query });
+      handleOpenChange(false);
+      pushInApp(router, target);
```
[the remark this answers is not in the document]
Comment (functionstackx) [reply packages/app/src/components/ui/command-palette.tsx]: Fixed in 275bec5 — selecting the current page is now a no-op (matching header links), and dashboard tab jumps carry the `unofficialruns` param the same way `TabNav` does. Covered by a new no-op test.

File: packages/app/src/lib/command-palette-items.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  const dashboard: PaletteNavItem[] = PRIMARY_TAB_ROUTES.map((route) => {
+    const key = route.key as DashboardRouteKey;
+    return {
+      id: `tab:${key}`,
+      group: 'dashboard',
+      href: route.path,
+      label: locale === 'zh' ? TAB_LABELS_ZH[key] : TAB_LABELS_EN[key],
+      keywords: locale === 'zh' ? TAB_LABELS_EN[key] : TAB_LABELS_ZH[key],
+    };
+  });
```
[the remark this answers is not in the document]
Comment (functionstackx) [reply packages/app/src/lib/command-palette-items.ts:108]: Fixed in 275bec5 — every dashboard destination now includes `dashboard` and `仪表板` in its keywords, so the header label users see hits these entries in both locales. Covered by a new registry test.
